woocommerce/woocommerce
# [Enhancement]:  Make Cart and Checkout Blocks the default experience on WooCommerce installation

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



## Changes proposed in this Pull Request:




In this PR we are introducing Cart and Checkout Blocks as default Cart and Checkout page content on WooCommerce first time activation, and through page creation on WooCommerce > Status > Tools > Create Pages.

Closes #40865
Closes #35029

## Developer Note: Deprecating `woocommerce_cart_shortcode_tag` and `woocommerce_checkout_shortcode_tag`

We are deprecating two filters, `woocommerce_cart_shortcode_tag` and `woocommerce_checkout_shortcode_tag`. These filters are no longer recommended for use and will be removed in future versions of WooCommerce.

#### Deprecated Filters

1. **`woocommerce_cart_shortcode_tag`**

   - Description: This filter allowed developers to modify the shortcode tag used for the WooCommerce cart.
   - Deprecated since: WooCommerce 8.3.0
   - Replacement: Developers should now use the `woocommerce_create_pages` filter instead to customize the cart page content.

2. **`woocommerce_checkout_shortcode_tag`**

   - Description: This filter allowed developers to modify the shortcode tag used for the WooCommerce checkout.
   - Deprecated since: WooCommerce 8.3.0
   - Replacement: Developers should now use the `woocommerce_create_pages` filter instead to customize the checkout page content.

#### Alternative: `woocommerce_create_pages` Filter

The `woocommerce_create_pages` filter provides a unified way to customize the default content for various WooCommerce pages, including the cart and checkout pages. By using this filter, you can set your own shortcode tag content for these pages, ensuring a consistent and cohesive approach.

Here's an example of how to use the `woocommerce_create_pages` filter to set custom shortcode tags for the cart and checkout pages:

```php
function customize_woocommerce_pages($pages) {
    $pages['cart']['content'] = '[custom_cart_shortcode]';
    $pages['checkout']['content'] = '[custom_checkout_shortcode]';
    return $pages;
}

add_filter('woocommerce_create_pages', 'customize_woocommerce_pages');
```


## How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Setup a new WooCommerce store w/ a blocks theme (TT3, TT4...)
2. Verify that Cart and Checkout default pages contain the Blocks version for Cart and Checkout, respectively
3. Complete a test order and verify everything works as expected and that the Car, Checkout and Order Confirmation steps use the new block-based experience

#### Testing Tools > Create pages functionality
1. Delete the Cart and Checkout pages, and empty the trash
2. Install and activate [WooCommerce Blocks v11.4.0+](https://github.com/woocommerce/woocommerce-blocks/releases) if the bundled version is lower. (you can check Blocks version on WooCommerce > Status)
3. Go to WooCommerce > Status > Tools and click `Create pages`
4. Verify that Cart and Checkout default pages contain the Blocks version for Cart and Checkout, respectively
5. Complete a test order and verify everything works as expected and that the  Car, Checkout and Order Confirmation steps use the new block-based experience

#### Testing Tooltips updates
1. Go to WooCommerce > Settings > Advanced
2. On the Cart and Checkout pages help tooltips verify the following text is displayed on hover

<img width="354" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/a859fc9f-b08a-4e22-9dfe-c55687a1faa2">
<img width="352" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/2bf9637e-6836-422c-adbc-939e79552f57">

#### Test installations through WP-CLI (Dev only)
1. Get a new env ready to install WP/WC using WP-CLI. You can use this [quick docker setup](https://github.com/woocommerce/woocommerce/files/13310898/new-test-env.zip)
2. Using WP-CLI set up WC using the following command sequence (alter args to suit your case)

```
wp config create --dbname=wordpress --dbuser=wordpress --dbpass=wordpress --dbhost=db  --force 
wp core install --url=http://localhost --title='WP Staging' --admin_user=admin --admin_password=pass --admin_email=admin@wp.loc
wp option set blog_public 0 

# change the version or install it from built zip, depending on your testing context. Here is an example for RC2
wp plugin install woocommerce --version=8.3.0-rc.2 --activate

wp plugin install wordpress-importer  --activate
wp import ./wp-content/plugins/woocommerce/sample-data/sample_products.xml --authors=skip

# make sure TT4 is active
wp theme activate twentytwentyfour 
```
3. Open the configured store on your browser
4. Complete a test order and verify everything works as expected and that the Car, Checkout and Order Confirmation steps use the new block-based experience



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Cart and Checkout Blocks are now the default checkout experience on WooCommerce
#### Comment 

</details>
